### PR TITLE
Issue 155: Update pager 0.0.x

### LINKED ISSUE #155: Update pager to follow Atomium changes
Atomium >= 2.27

### PR DESCRIPTION
**Pull request by aritomelo**
Fix #155

### CODE REVIEW

File: templates/pager/pager.component.inc [hunk 1]
```diff
@@ -9,6 +9,7 @@
  * Implements hook_preprocess_hook().
  */
 function ec_europa_preprocess_pager(array &$variables, $hook) {
+
```
Comment (drupol) [inline templates/pager/pager.component.inc]: Unrelated changes - empty blank line.

File: templates/pager/pager.component.inc [hunk 2]
```diff
@@ -17,69 +18,120 @@ function ec_europa_preprocess_pager(array &$variables, $hook) {
...
 
-  // Update text to use Previous instead of previous.
-  if (isset($items['pager_previous']['pager_previous'])) {
-    $items['pager_previous']['pager_previous']['#text'] = t('‹ Previous');
+  // Logic to append first ELLIPSIS.
+  // Case the pager_first is shown do not display pager_1.
+  $pager_links = _pager_get_item_keys($items, 'pager_link');
+  $pager_first = _pager_get_item_keys($items, 'pager_first');
+  $pager_current = _pager_get_item_keys($items, 'pager_current');
+  if (!empty($pager_first) && !empty($pager_links)) {
```
Comment (drupol) [inline templates/pager/pager.component.inc]: Maybe we should add one more check.

Comment (drupol) [inline templates/pager/pager.component.inc]:
```suggestion
  if (!empty($pager_first) && !empty($pager_links) && $ellipsis !== NULL) {
```

File: templates/pager/pager.component.inc [hunk 3]
```diff
@@ -113,16 +165,17 @@ function ec_europa_preprocess_pager(array &$variables, $hook) {
 
   // If there is an ellipsis and if the current element is within 4 elements
   // before the end of the links array.
-  if ((count($links) - $index_current < 4) && $has_ellipsis === TRUE) {
+  if (((count($links) - $index_current < 4)) && $has_ellipsis === TRUE) {
```
Comment (drupol) [inline templates/pager/pager.component.inc]:
```suggestion
  if (((count($links) - $index_current) < 4) && $has_ellipsis === TRUE) {
```

File: templates/pager/pager.component.inc [hunk 4]
```diff
@@ -152,8 +215,78 @@ function ec_europa_preprocess_pager(array &$variables, $hook) {
...
-    if (isset($classMap[$base])) {
-      $item['#attributes']['class'][] = 'ecl-pager__item--' . $classMap[$base];
+    if (isset($classMap[$item_type])) {
+      $item['#attributes']['class'][] = 'ecl-pager__item--' . $classMap[$item_type];
+    }
+  }
+}
+
+/**
+ * Get the pager item/s by pagers type.
```
Comment (drupol) [inline templates/pager/pager.component.inc]:
```suggestion
 * Get the pager items by pager type.
```

File: templates/pager/pager.component.inc [hunk 5]
```diff
@@ -152,8 +215,78 @@ function ec_europa_preprocess_pager(array &$variables, $hook) {
...
+  $pager_type_key = _pager_get_item_keys($items, $pager_type);
+  $pager_type_items = array();
+
+  foreach ($pager_type_key as $key) {
+    if (isset($items[$key])) {
+      $pager_type_items[] = $items[$key];
+    }
+  }
+
+  return $pager_type_items;
```
Comment (drupol) [inline templates/pager/pager.component.inc]: I think we can use `array_intersect_key()` here and remove the `foreach` loop.

File: templates/pager/pager.component.inc [hunk 6]
```diff
@@ -152,8 +215,78 @@ function ec_europa_preprocess_pager(array &$variables, $hook) {
...
+ * @param array $items
+ *   The list of pager items.
+ * @param string $pager_type
+ *   The pager type designation: pager_first, pager_last, etc.
+ *
+ * @return array
+ *   A list of items belonging to the same pager type.
+ */
+function _pager_get_items_by_pager_type(array $items, $pager_type) {
+  $pager_type_key = _pager_get_item_keys($items, $pager_type);
```
Comment (drupol) [inline templates/pager/pager.component.inc]:
```suggestion
  $pager_type_keys = _pager_get_item_keys($items, $pager_type);
```

File: templates/pager/pager.component.inc (hunk 5 above)
Comment (drupol) [inline templates/pager/pager.component.inc]:
```suggestion
  return array_intersect_key($items, array_flip($pager_type_keys));
```

File: templates/pager/pager.component.inc [hunk 7]
```diff
@@ -152,8 +215,78 @@ function ec_europa_preprocess_pager(array &$variables, $hook) {
...
+function _pager_get_item_keys(array $items, $pager_type) {
+  $keys = array();
+
+  foreach ($items as $index => $item) {
+    if ($item['#pager_item_type'] === $pager_type) {
+      $keys[] = $index;
     }
   }
+
+  return $keys;
```
Comment (drupol) [inline templates/pager/pager.component.inc]:
```suggestion
  return array_filter($items, function ($item) use ($pager_type) {return $item['#pager_item_type'] === $pager_type;});
```

File: templates/pager/pager.component.inc [hunk 8]
```diff
@@ -152,8 +215,78 @@ function ec_europa_preprocess_pager(array &$variables, $hook) {
...
+ * @param string $pager_type
+ *   The pager type designation: pager_first, pager_last, etc.
+ *
+ * @return array
+ *   The array with $pager_type key within $items.
+ */
+function _pager_get_item_keys(array $items, $pager_type) {
+  $keys = array();
+
+  foreach ($items as $index => $item) {
```
Comment (drupol) [inline templates/pager/pager.component.inc]: Remove the foreach loop and use the `array_filter()`.

File: templates/pager_next/pager_next.component.inc [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * @file
+ * Contains component file.
+ */
+
+/**
+ * Implements hook_preprocess_hook().
+ */
+function ec_europa_preprocess_pager_next(array &$variables, $hook) {
+  // Replace next by Next.
```
Comment (drupol) [inline templates/pager_next/pager_next.component.inc]: Remove comment or add it also in `pager_previous.component.inc`.

File: templates/pager/pager.component.inc (hunk 6 above)
Comment (drupol) [reply templates/pager/pager.component.inc]: When a function is returning an array, it's better to add a 's' to the variable. It makes more sense.

File: templates/pager/pager.component.inc (hunk 1 above)
Comment (aritomelo) [reply templates/pager/pager.component.inc]: ok.
